_posts/2021-04-28-#17331.md

```diff
@@ -0,0 +1,111 @@
+
+    - Maintaining financial privacy:  
+      Ideally, our transactions leak as little information about ourselves and
+      our business partners as possible.
+
+    - Reliably move the payment(s) to finalization:  
+      Reduce the chance of transactions being conflicted and transfer value in a
+      limited time-frame
+
+- Using unconfirmed inputs can make transactions unreliable. When a foreign
+  input is used, the parent transaction may time out or get replaced. Even using
+  self-sent unconfirmed funds may delay the new transaction if the parent
+  transaction has an extensive ancestry, is extraordinarily large, or used a
+  lower feerate than targeted for the child transaction.
+
+- When a new transaction is built, the target feerate is provided as an input.
+
+- When a UTXO is selected as an input, the size of the transaction is increased
+  which in turn increases the necessary fees to maintain the targeted feerate.
+  We can preempt the cost added by the input by considering each UTXO at its
+  _effective value_, its value reduced by the UTXO's input cost at the given
+  feerate.
+
+- Using _effective values_ in coin selection means that we can freely add and
+  remove inputs until we have have raised enough funds to pay for the
+  transaction's fixed costs: the payment amounts, the transaction header and the
+  output data. Whatever final input set we settle for, the increased transaction
+  size and input costs have already been accounted for.
+
+- The _knapsack solver_ sorts the available UTXOs by effective value, and then
+  combines candidate input sets by randomly selecting or skipping inputs until
+  it has sufficient funds for the transaction. When it finds a viable solution,
+  it keeps the solution if it produces a smaller change than the prior stored
+  solution. Then, the algorithm deselects the last input and continues
+  traversing the UTXO list as described, trying to converge on the smallest
+  change output above `MIN_CHANGE`. This random process is repeated up to 1,000
+  times.
+
+- The _Branch and Bound solver (BnB)_ deterministically searches the complete
+  combination space. It performs a depth-first search on a binary tree where
+  each branch represents the inclusion or exclusion of a UTXO. It explores each
+  inclusion branch first, and backtracks whenever a subtree cannot yield a
+  solution. It returns the first discovered input set that is an "exact match"
+  for the funding requirement of the transaction. A resulting transaction avoids
+  the creation of a change output, but the BnB algorithm is not guaranteed to
+  find a solution even when there are sufficient funds.
+
+- An
+  [OutputGroup](https://github.com/bitcoin-core-review-club/bitcoin/blob/4ac1adda9914d845aaea5804af4801ffec53c701/src/wallet/coinselection.h#L72)
+  collects all UTXOs that were sent to the same public key.
+
+- This PR implements the use of _effective values_ across all different coin
```

Comment:
Also mention that the PR makes `SelectCoinsMinConf` try both strategies (i.e. BnB then Knapsack if needed) instead of taking a bool.

Reply:
Added a bullet point!